.github/workflows/deps.yml

```diff
@@ -0,0 +1,45 @@
+name: Dependency Changes
+
+# Trigger on PRs.
+on:
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  # Compare dependencies before and after this PR.
+  dependencies:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    strategy:
+      fail-fast: true
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: stable
+          cache-dependency-path: "**/*go.sum"
+
+      # Run the commands to generate dependencies before and after and compare.
+      - name: Compare dependencies
+        run: |
+          BEFORE="$(mktemp -d)"
+          AFTER="$(mktemp -d)"
+
+          cp scripts/gen-deps.sh /tmp/  # TODO: delete
+          scripts/gen-deps.sh "${AFTER}"
+          git checkout origin/master
+          cp /tmp/gen-deps.sh scripts/  # TODO: delete
+          scripts/gen-deps.sh "${BEFORE}"
```

Comment:
Would it make sense to add a `grpc-dir` parameter to `gen-deps.sh` that it would use if provided and if not provided, it would simply `cd` to `"${SCRIPTS_DIR}/.."`.

If we have that option, then we could replace this block with
- copy the gen-deps.sh from current branch into /tmp
- run /tmp/gen-deps.sh on `grpc-dir`
- checkout master
- run /tmp/gen-deps.sh on `grpc-dir`

Reply:
The copying is temporary for the purposes of this PR only.  `origin/master` will have this script as soon as I submit this PR.

Reply:
Ack.